### ISSUE TITLE
Why the server keeps dying bruh?

### ISSUE THREAD

**Issue by ice051128** [RobinLinus/snapdrop#527]
The server literally has a 25% downtime over one month like wtf

https://www.saashub.com/snapdrop-status

**Comment by jflattery** [comment]
[ice051128](https://github.com/ice051128), I would recommend you formally request a refund.

**Comment by MattRCole** [comment]
While I don't agree with the tone of this issue at all, the question raised is a valid one: The website has been down quite a bit recently.

- Is this due to cost of hosting or because of server errors? Or is it something else entirely?

- Is there anything that members of the community can do to help?

I gotta say, there's nothing quite like losing access something to help you realize how much you rely on it.

Thank you for the work that everyone puts into this to keep it going.

**Comment by MattRCole** [comment]
Found the answer: https://github.com/RobinLinus/snapdrop/issues/515

**Comment by imh1kiko** [comment]
It's down yet again.
This is getting a bit too unreliable for me, so as said in #515, anyone who wants, can locally [host their own](https://github.com/JustSch/node-snapdrop-electron/releases).

**Comment by jflattery** [comment]
As also stated in the mentioned thread, the project is being re-written in rust, and I suspect that will make a major difference in uptime and performance.

**Comment by codestation** [comment]
Selfhosting is the way to go. I put the docker image referenced in the wiki behind a Traefik reverse proxy in 5 minutes.

```
docker run -d -p 8080:80 linuxserver/snapdrop
```

The memory footprint is small enough (~20 MB idle) to put it in a cheap VPS.

**Comment by Bellisario** [comment]
> Maybe [Snapdrop for Android](https://play.google.com/store/apps/details?id=com.fmsys.snapdrop) causes a problem because there are about a Million clients permanently connecting to the server and thus dos'ing it?

Yes, this can be really possible… consider my node-snapdrop project seems to be active about 14 hours per day and I noticed that’s crashing perfectly the same as Snapdrop (web socket error crashing the server)

**Comment by dcw131313** [comment]
I opened up Snapdrop an hour ago and transferred over 364 photos from my android to my dell laptop and it took just a few minutes and the program worked great, then a phone call came and I answered it without closing the app and now it says I have to verify my wifi connection.  I am connected and my speeds are 140 download and 136 upload.

**Comment by danielehrhardt** [comment]
[Disable iCloud Private Relay 
](https://filedrop.codext.de/)

**Comment by fm-sys** [comment]
> Maybe [Snapdrop for Android](https://play.google.com/store/apps/details?id=com.fmsys.snapdrop) causes a problem because there are about a Million clients permanently connecting to the server and thus dos'ing it?

Hi, I want to give some more insights about Snapdrop for Android:
Currently, Snapdrop for Android is installed on ~500k active devices. However, the app does only cause traffic if users actively open the app. Thinking about myself, let's say I'm using snapdrop for maybe 30min a week which would make 0.3% of the total time. If everyone uses Snapdrop that much as I do, there would be 1500 parallel users in average. Assuming that only 10% of all installations are using the app regularly, we are at 150 parallel users. By far, not enough to crash servers.
Furthermore, from my experience, users only install the app cause they have heard from snapdrop.net and search for an android client. Even if such a client would not exist, they would use the website anyway.

The main problem is, that more and more Youtubers start recommending snapdrop. Counting all views together, probably several million people have already heard from snapdrop. Not to forget, that youtube is by far not the only way to make people aware of snapdrop. The greatness of snapdrop is also its biggest enemy ;-)
